thetis/optimisation.py

```diff
@@ -20,20 +23,25 @@ class UserExportManager(ExportManager):
     """
     ExportManager for user provided functions (not necessarily known to Thetis)
 
-    If function.name() is in field_defs.field_metadata, the filename and shortname are
-    obtained there. If the function.name() is unknown, both are based on function.name()
-    directly (with an optional additional filename_prefix). Filenames and shortnames
-    can be overruled by the shortnames and filenames arguments."""
+    In the standard :class:`.ExportManager` all provided functions need to have standard names
+    present in :py:data:`.field_metadata`. Here, any functions can be provided. If function.name() is in
+    :py:data:`.field_metadata`, the standard filename and shortname  are used.
+    If the function.name() is unknown, both are based on function.name()
+    directly (with an optional additional filename_prefix). Filenames and
+    shortnames can be overruled by the shortnames and filenames arguments."""
```

Comment:
We need to generalize the `field_metadata` mechanism, so that users can easily add new fields with all the metadata.

Reply:
Agreed. This was a a bit of a workaround. Related, it would also be good to have a discussion about how to do advection of an arbitrary number of scalars, not just salinity and temperature.

Reply:
Yes, we'll definitely need that soon.